central/debug/service/zip.go

```diff
@@ -0,0 +1,35 @@
+package service
+
+import (
+	"archive/zip"
+	"io"
+	"path"
+	"time"
+
+	"github.com/stackrox/rox/pkg/k8sintrospect"
+)
+
+var now = func() time.Time {
```

Comment:
Alternative proposal:

```
func zipWriterWithCurrentTimestamp(zipWriter *zip.Writer, fileName string) (io.Writer, error) {
	return zipWriterWithCurrentTimestampUseNowFunc(zipWriter, fileName, func() time.Time { 
		return time.Now() 
	})
}

func zipWriterWithCurrentTimestampUseNowFunc(zipWriter *zip.Writer, fileName string, nowFunc func() time.Time ) (io.Writer, error) {
	header := &zip.FileHeader{
		Name:     fileName,
		Method:   zip.Deflate,
		Modified: now(),
	}
	return zipWriter.CreateHeader(header)
}
```

And using the second function for testing

Reply:
I believe it's better to have smaller dependencies. Function that returns just a current time gives me better coverage than mocking a whole function.

Reply:
Ideally, I think we would've had a struct calling these methods and that's where we would pass `now()` function, but that is too big of a refactoring for such a case. I won't push - let's keep it as it is

Reply:
We can fix it in a separated PR as this will means that most of functions will be struct functions as we need `now()` to be somewhere defined

Reply:
Yea, let's not fix it here